decimal.go

```diff
@@ -1102,6 +1104,8 @@ func (z *Big) SetString(s string) (*Big, bool) {
 				return nil, false
 			}
 			err = nerr.Err
+		} else if z.compact == 0 {
```

Comment:
This should only run if err == nil. If err != nil and the return 3 lines above isn't triggered, then the unscaled block below will run which does its own form calculation. If this weren't an `else if`, then z.form would be set to zero on any error that also returns 0 as its first argument. Reading the code in ParseInt a RangeErr does also return a non-zero first argument, so it appears that this could be an else if or a new if and the results are the same.